File: src/main/java/storebackend/dto/DhlFindParcelRequest.java
package storebackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Request: Paket suchen
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DhlFindParcelRequest {
    /**
     * Roher Tracking-Code (wird serverseitig normalisiert)
     */
    private String trackingCode;
}

File: mobile-flutter-poc/lib/models/dhl_find_parcel_request.dart
/// Schlankes Flutter-Gegenstueck zu `DhlFindParcelRequest.java`
/// (storebackend.dto) - Request fuer `POST
/// /api/stores/{storeId}/dhl/parcels/find` (siehe `DhlController.findParcel`).
///
/// Bewusst nur das eine Feld, das der Endpoint tatsaechlich erwartet - der
/// rohe Tracking-Code wird serverseitig normalisiert, keine
/// Client-Normalisierung noetig.
class DhlFindParcelRequest {
  const DhlFindParcelRequest({required this.trackingCode});

  final String trackingCode;

  Map<String, dynamic> toJson() => {'trackingCode': trackingCode};
}
